manifests/init.pp

```diff
@@ -32,11 +50,15 @@
     create_resources('systemd::service_limits', $service_limits)
   }
 
-  if $manage_resolved {
+  if $manage_resolved and $facts['systemd_internal_services'] and $facts['systemd_internal_services']['systemd-resolved.service'] {
```

Comment:
Maybe a bit late to the party but if ```manage_resolved``` is true and the system can't use it, should that be an error?

Reply:
this would add another layer, not sure if this is actually needed. Is it safe to assume that only people that know their systems try to enable it? (Or at least know if they run systemd or not)

Reply:
Also using the module would be a bit more complicated/requires more logic. Currently I apply it to _all_ servers in our org, no matter if its a freebsd, old centos or something with systemd. Fail safely vs fail hard

Reply:
On the other hand, if you don't manage this you might silently not manage something when you expect it to. Makes me wonder if it should be a tristate: off, auto and on where on has checks if it's actually enabled.

Reply:
I thought about this, but unsure of the needed complexity is worth it